Add `createNamedParameter` to `QueryBuilder` (#11528)
doctrine/doctrine2: docs/en/reference/query-builder.rst

      ->add('from', new Expr\From('User', 'u'))
      ->add('where', new Expr\Comparison('u.id', '=', '?1'))
      ->add('orderBy', new Expr\OrderBy('u.name', 'ASC'));

Binding Parameters to Placeholders
----------------------------------

It is often not necessary to know about the exact placeholder names

> during the building of a query. You can use two helper methods

Reviewer: ```suggestion
when building a query. You can use two helper methods
```
